lib/cucumber/formatter/json.rb

```diff
@@ -105,7 +105,7 @@ def same_feature_as_previous_test_case?(test_case)
       end
 
       def first_step_after_background?(test_step)
-        @in_background && test_step.location.lines.max >= @test_case_hash[:line]
+        @in_background && test_step.location.file == @feature_hash[:uri] && test_step.location.lines.max >= @test_case_hash[:line]
```

Comment:
Nice catch.
How did you find it? Would it be possible and not too difficult to add a test?

Reply:
We use `cucumber-rails` which pulls in `capybara`, and capybara I guess has it's own `before` hooks.

Reply:
If you have a suggestion for testing this, I'm all ears, but I'm not that intimately familiar with the library

Reply:
I'll take a look